apps/cowswap-frontend/src/modules/swap/updaters/SmartSlippageUpdater.ts

```diff
@@ -42,12 +44,115 @@ export function SmartSlippageUpdater() {
 
       return response.slippageBps
     },
-    SWR_OPTIONS
+    SWR_OPTIONS,
   ).data
 
+  // TODO: remove v1
+  const tradeSizeSlippageBpsV1 = useSmartSlippageFromFeePercentage()
+  const tradeSizeSlippageBps = useSmartSlippageFromFeePercentageV2()
+
+  useEffect(() => {
+    // Add both slippage values, when present
+    const slippage = tradeSizeSlippageBps + (bffSlippageBps || 0)
+
+    setSmartSwapSlippage(slippage)
+  }, [bffSlippageBps, setSmartSwapSlippage, tradeSizeSlippageBps])
+
+  // TODO: remove before merging
   useEffect(() => {
-    setSmartSwapSlippage(typeof slippageBps === 'number' ? slippageBps : null)
-  }, [slippageBps, setSmartSwapSlippage])
+    console.log(`SmartSlippageUpdater`, {
+      granularSlippage: tradeSizeSlippageBpsV1,
+      fiftyPercentFeeSlippage: tradeSizeSlippageBps,
+      bffSlippageBps,
+    })
+  }, [tradeSizeSlippageBpsV1, tradeSizeSlippageBps])
 
   return null
 }
+
+const FEE_MULTIPLIER_FACTOR = new Fraction(15, 10) // 50% more fee, applied to the whole value => 150% => 15/10 in fraction
+const ONE = new Fraction(1)
+const ZERO = new Fraction(0)
+
+/**
+ * Calculates smart slippage in bps, based on quoted fee
+ *
+ * Apply a multiplying factor to the fee (e.g.: 50%), and from there calculate how much slippage would be needed
+ * for the limit price to take this much more fee.
+ * More relevant for small orders in relation to fee amount, negligent for larger orders.
+ */
+function useSmartSlippageFromFeePercentageV2(): number {
+  const { trade } = useDerivedSwapInfo() || {}
+  const { fee, inputAmountWithFee, inputAmountWithoutFee, tradeType } = trade || {}
+  const { feeAsCurrency } = fee || {}
+
+  const percentage = useMemo(() => {
+    if (!inputAmountWithFee || !inputAmountWithoutFee || !feeAsCurrency || tradeType === undefined) {
+      return ZERO
+    }
+
+    if (tradeType === TradeType.EXACT_INPUT) {
+      // sell
+      // 1 - (sellAmount - feeAmount * 1.5) / (sellAmount - feeAmount)
+      // 1 - (inputAmountWithoutFee - feeAsCurrency * 1.5) / inputAmountWithFee
+      return ONE.subtract(
+        inputAmountWithoutFee
+          .subtract(feeAsCurrency.multiply(FEE_MULTIPLIER_FACTOR))
+          // !!! Need to convert to fraction before division to not lose precision
+          .asFraction.divide(inputAmountWithFee.asFraction),
+      )
+    } else {
+      // buy
+      // (sellAmount + feeAmount * 1.5) / (sellAmount + feeAmount) - 1
+      // (inputAmountWithFee + feeAsCurrency * 1.5) / inputAmountWithFee - 1
+      return (
+        inputAmountWithFee
+          .add(feeAsCurrency.multiply(FEE_MULTIPLIER_FACTOR))
+          // !!! Need to convert to fraction before division to not lose precision
+          .asFraction.divide(inputAmountWithFee.asFraction)
+          .subtract(ONE)
+      )
+    }
```

Comment:
Core logic for calculating slippage based on fee

Reply:
The logic seems to match [the doc](https://docs.google.com/spreadsheets/d/1WtlJ8qKh-DlZy8xS7pyXZkd3MdXysT5J2T4aUHhDd_Q/edit?gid=1874413274#gid=1874413274).